src/DataProtection/DataProtection/src/KeyManagement/IKeyManagerWithDeletion.cs

```diff
@@ -0,0 +1,51 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+
+namespace Microsoft.AspNetCore.DataProtection.KeyManagement;
+
+/// <summary>
+/// The basic interface for performing key management operations.
+/// </summary>
+/// <remarks>
+/// Instantiations of this interface are expected to be thread-safe.
+/// </remarks>
+public interface IKeyManagerWithDeletion : IKeyManager
+{
+    /// <summary>
+    /// Indicates whether this key manager supports key deletion.
+    /// </summary>
+    /// <remarks>
+    /// Deletion is stronger than revocation.  A revoked key is retained and can even be (forcefully) applied.
+    /// A deleted key is indistinguishable from a key that never existed.
+    /// </remarks>
+    bool CanDeleteKeys { get; }
```

Comment:
@halter73 I kept this because deletion will fail if the underlying repository doesn't support it and I thought it made sense to provide a way to query for that, rather than just trying and failing.

Reply:
I was going to send mail with the final shape, but discussing it Monday might be quicker.